static/src/javascripts/projects/common/modules/identity/consent-journey.js

```diff
@@ -58,12 +81,35 @@ const showJourneyAlert = (journeyEl: HTMLElement): void => {
         formEl.addEventListener('submit', ev => {
             if (ev.isTrusted) {
                 ev.preventDefault();
-                Promise.all([
-                    shouldDisplayNewsletterWarning(journeyEl),
-                    shouldDisplayConsentWarning(journeyEl),
-                ]).then(([emptyNewsletters, emptyMarketing]) => {
-                    if (emptyNewsletters && emptyMarketing) {
-                        showModal('confirm-consents');
+                const uncheckBoxesPromise = uncheckedBoxes(journeyEl, [
+                    'email',
+                    'marketing-consents',
+                ]);
+                uncheckBoxesPromise.then(unchecked => {
+                    if (unchecked.length > 0) {
+                        const consentNames = unchecked.map(
+                            uncheckedbox =>
+                                uncheckedbox.parentElement.querySelector(
+                                    '.manage-account__switch-title'
+                                ).innerText
+                        );
+                        getModal('confirm-consents').then(modalEl =>
```

Comment:
almost there! you can still use `showModal`, and you don't have to wait until the modal is displayed to render the email list, so this can become

```
getModal('confirm-consents').then(modalEl =>
  if (consentNames) {
    insertNewsletterReminders(
      nextModalEl,
      consentNames
    );
  }
});
showModal('confirm-consents');
```
and they can run in parallel!